cpp/example_code/dynamodb/query_items.cpp

```diff
@@ -1,130 +1,159 @@
-            const Aws::Vector<Aws::Map<Aws::String, Aws::DynamoDB::Model::AttributeValue>>& items = result.GetResult().GetItems();
-            if(items.size() > 0) 
-            {
-                std::cout << "Number of items retrieved from Query: " << items.size() << std::endl;
+            const Aws::Vector<Aws::Map<Aws::String, Aws::DynamoDB::Model::AttributeValue>> &items = outcome.GetResult().GetItems();
+            if (!items.empty()) {
+                std::cout << "Number of items retrieved from Query: " << items.size()
+                          << std::endl;
                 //Iterate each item and print
-                for(const auto &item: items)
-                {
-                std::cout << "******************************************************" << std::endl;
-                // Output each retrieved field and its value
-                for (const auto& i : item)
-                    std::cout << i.first << ": " << i.second.GetS() << std::endl;
+                for (const auto &item: items) {
+                    std::cout
+                            << "******************************************************"
+                            << std::endl;
+                    // Output each retrieved field and its value
+                    for (const auto &i: item)
+                        std::cout << i.first << ": " << i.second.GetS() << std::endl;
                 }
             }
-
-            else
-            {
-                std::cout << "No item found in table: " << table << std::endl;
+            else {
+                std::cout << "No item found in table: " << tableName << std::endl;
             }
+
+            exclusiveStartKey = outcome.GetResult().GetLastEvaluatedKey();
         }
-        else
-        {
-            std::cout << "Failed to Query items: " << result.GetError().GetMessage();
+        else {
+            std::cerr << "Failed to Query items: " << outcome.GetError().GetMessage();
+            result = false;
+            break;
         }
-        // snippet-end:[dynamodb.cpp.query_items.code]
+    } while (!exclusiveStartKey.empty());
+
+    return result;
+}
+// snippet-end:[dynamodb.cpp.query_items.code]
+
+/*
+ *  main function
+ *
+ *  Usage: 'run_query_items <table_name> <partition_key> <partition_value> [projection_expression]'
+ *
+ *  Prerequisites: A pre-populated DynamoDB table.
+ *
+ *  Instructions for populating a table with sample data can be found at:
+ *  https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SampleData.html
+ *
+ */
+
+#ifndef TESTING_BUILD
+
+int main(int argc, char **argv) {
+    if (argc < 4) {
+        std::cout << R"(
+Usage:
+    run_query_items <table_name> <partition_key> <partition_value> [projection_expression]
+Where:
+    table_name - the table to get an item from
+    partition_key  - the partition key attribute of the table
+    partition_value  - the partition key value to query
+    [projection_expression] - the projection expression

```

Comment:
table_name - The table to get an item from.
    partition_key  - The partition key attribute of the table.
    partition_value  - The partition key value to query.
    [projection_expression] - The projection expression.